scripts/quests/windurst/Mihgo_s_Amigo.lua

```diff
@@ -0,0 +1,151 @@
+                player:getQuestStatus(xi.questLog.WINDURST, xi.quest.id.windurst.AS_THICK_AS_THIEVES) ~= xi.questStatus.QUEST_ACCEPTED and
+                lostForWords == 0 and -- Player cannot accept quest during Lost For Words Mission
+                not player:hasItem(xi.item.MARAUDERS_KNIFE)
+        end,
+
+        [xi.zone.WINDURST_WOODS] =
+        {
+            ['Nanaa_Mihgo'] =
+            {
+                onTrigger = function(player, npc)
+                    if player:getQuestStatus(xi.questLog.WINDURST, xi.quest.id.windurst.CRYING_OVER_ONIONS) == xi.questStatus.QUEST_ACCEPTED then
+                        return quest:progressEvent(81) -- come to apologize
+                    else
+                        return quest:progressEvent(80)
+                    end
+                end,
+            },
+
+            onEventFinish =
+            {
+                [80] = function(player, csid, option, npc)
+                    return quest:begin(player)
+                end,
+
+                [81] = function(player, csid, option, npc)
+                    return quest:begin(player)
+                end,
+            },
+        },
+    },
+
+    {
+        -- QUEST ACCEPTED
+        check = function(player, status, vars)
+            return status == xi.questStatus.QUEST_ACCEPTED
+        end,
+
+        [xi.zone.WINDURST_WOODS] =
+        {
+            ['Bopa_Greso'] = quest:event(84, 0, xi.item.YAGUDO_BEAD_NECKLACE), -- Migho's Amigo hint dialog
+
+            ['Cha_Lebagta'] = quest:event(85, 0, xi.item.YAGUDO_BEAD_NECKLACE), -- Migho's Amigo hint dialog
+
+            ['Gioh_Ajihri'] = quest:event(87),
+
+            ['Mheca_Khetashipah'] = quest:event(83),
+
+            ['Nanaa_Mihgo'] =
+            {
+                onTrade = function(player, npc, trade)
+                    if npcUtil.tradeHasExactly(trade, { { xi.item.YAGUDO_BEAD_NECKLACE, 4 } }) then
+                        if player:getQuestStatus(xi.questLog.WINDURST, xi.quest.id.windurst.CRYING_OVER_ONIONS) == xi.questStatus.QUEST_ACCEPTED then
+                            return quest:progressEvent(92, { [0] = xi.settings.main.GIL_RATE * gilPerTrade }) -- apology accepted
+                        else
+                            return quest:progressEvent(88, { [0] = xi.settings.main.GIL_RATE * gilPerTrade })
+                        end
+                    end
+                end,
+
+                onTrigger = function(player, npc)
+                    return quest:event(82)
+                end,
+            },
+
+            ['Wani_Casdohry'] = quest:event(86, 0, xi.item.YAGUDO_BEAD_NECKLACE), -- Migho's Amigo hint dialog
+
+            onEventFinish =
+            {
+                [88] = function(player, csid, option, npc)
+                    if quest:complete(player) then
+                        player:confirmTrade()
+                        player:addGil(xi.settings.main.GIL_RATE * gilPerTrade)
```

Comment:
Is there a reason quest.reward value isn't being set and used here? That handles the gil rate config

Reply:
Yep! The CS itself includes the gil received message so by setting it in the rewards table I was getting double messages.

Reply:
There are a few others that have the gil reward baked into the CS dialog itself; see
https://github.com/LandSandBoat/server/blob/base/scripts/quests/sandoria/The_Sweetest_Things.lua#L91-L95